Checkout/Block/Form/Payments.php

```diff
@@ -141,12 +148,52 @@ public function getIframeUrl()
     private function getStyles()
     {
         $styles = Mage::getModuleDir('data', 'Bold_Checkout') . DS . 'form/payments/styles.json';
-        // @phpcs:disable MEQP1.Security.DiscouragedFunction.Found
         if (file_exists($styles)) {
             return json_decode(file_get_contents($styles), true);
         }
-        // @phpcs:enable MEQP1.Security.DiscouragedFunction.Found
-
         return null;
     }
+
+    /**
+     * Refresh order on Bold side before render PIGI iframe.
+     *
+     * @return void
+     * @throws Mage_Core_Exception
+     */
+    private function refreshOrder()
+    {
+        $session = Mage::getSingleton('checkout/session');
+        $boldCheckoutData = $session->getBoldCheckoutData();
+        if (!$boldCheckoutData) {
+            return;
+        }
+        $boldEmailAddress = $boldCheckoutData->data->application_state->customer->email_address;
+        $email = $session->getQuote()->getBillingAddress()->getEmail();
+        $firstname = $session->getQuote()->getBillingAddress()->getFirstname();
+        $lastname = $session->getQuote()->getBillingAddress()->getLastname();
+        if ($this->customerIsGuest() && !$boldEmailAddress && $email && $firstname && $lastname) {
```

Comment:
@nmalevanec  Is this required? Couldn't we just in all scenario's make a request to /refresh api. This should then fetch all shipping, billing, and customer information.

Reply:
Removed. Ticket created for guest customer checkout  - https://boldapps.atlassian.net/browse/INTER-2545